libdnf/hy-goal.c

```diff
@@ -1381,7 +1381,7 @@ hy_goal_get_reason(HyGoal goal, DnfPackage *pkg)
     //solver_get_recommendations
     if (goal->solv == NULL)
         return HY_REASON_USER;
-    Id info;
+    Id info, pkg_id;
     int reason = solver_describe_decision(goal->solv, dnf_package_get_id(pkg), &info);
```

Comment:
Small optimalization :-) Call `dnf_package_get_id(pkg)` only once.

`const Id pkg_id = dnf_package_get_id(pkg);`
`Id info;`
`int reason = solver_describe_decision(goal->solv, pkg_id, &info);`

and remove line 1399 `pkg_id = dnf_package_get_id(pkg);`